freegames/madlibs.py

```diff
@@ -0,0 +1,59 @@
+""" Mad Libs: Funny Story Creation Game.
+
+Exercises:
+
+1. Can you create functionality to replace the given story?
+2. Can you load the story and template from a file?
+3. Can you add additional tenses?
+4. Can you add flexibility in a loaded story template?
+5. Can you adjust for grammar? (past tense, etc.)
+
+"""
+
+story = "The [adj] [adj] [n] [v] over the [adj] [adj] [n]"
+# The quick black fox jumped over the lazy brown dog.
+
+word_type_counter = {}
+
+word_storage = {}
+
+word_type_key = {
+    "adj": "adjective",
+    "n": "noun",
+    "v": "verb",
+    "adv": "adverb"
+}
+
+def extract_word_types():
+    "Word types required extracted from story"
+    for w_type in word_type_key:
+        lookup = "[{}]".format(w_type)
+        word_type_counter[w_type] = story.count(lookup)
+
+def get_words():
+    "Get words from user"
+    for word in word_type_counter:
+        while word_type_counter[word] > 0:
```

Comment:
Instead of getting all the words by type, (like all the nouns, then all the verbs, etc.), can you query for the types in the order that they appear? Like for the example, it would be: give me an adjective, another adjective, a noun, a verb, an adjective, and a noun. I think it's more interesting if the user goes back and forth rather than doing each type individually.

Reply:
Agree! It would also be more straightforward. Ok. In progress.

Reply:
Ok. This is in place.